extensions/maven-profile/src/test/java/com/liferay/blade/extensions/maven/profile/BuildServiceCommandMavenTest.java

```diff
@@ -0,0 +1,117 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.liferay.blade.extensions.maven.profile;
+
+import com.liferay.blade.cli.BladeTestResults;
+import com.liferay.blade.cli.TestUtil;
+
+import java.io.File;
+
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+/**
+ * @author Christopher Bryan Boyd
+ */
+public class BuildServiceCommandMavenTest {
+
+	@Before
+	public void setUp() throws Exception {
+		_workspaceDir = temporaryFolder.newFolder("build", "test", "workspace");
+
+		_extensionsDir = temporaryFolder.newFolder(".blade", "extensions");
+	}
+
+	@Test
+	public void testMavenServiceBuilder() throws Exception {
+		String[] args = {"--base", _workspaceDir.getPath(), "init", "-P", "maven", "mavenworkspace"};
+
+		File mavenworkspace = new File(_workspaceDir, "mavenworkspace");
+
+		TestUtil.runBlade(mavenworkspace, _extensionsDir, args);
+
+		Assert.assertTrue(mavenworkspace.exists());
+
+		args = new String[] {"--base", mavenworkspace.getPath(), "create", "-t", "service-builder", "sb1"};
+
+		TestUtil.runBlade(mavenworkspace, _extensionsDir, args);
+		
+		args = new String[] {"--base", mavenworkspace.getPath(), "create", "-t", "service-builder", "sb2"};
+
+		TestUtil.runBlade(mavenworkspace, _extensionsDir, args);
+		
+		Path mavenworkspacePath = mavenworkspace.toPath();
+
+		
+		Path modulesPath = mavenworkspacePath.resolve("modules");
+
+		Path sb1SourcePath = Paths.get("sb1", "sb1-service", "src");
+
+		
+		sb1SourcePath = modulesPath.resolve(sb1SourcePath);
+		
+		Path sb2SourcePath = Paths.get("sb2", "sb2-service", "src");
+
+		
+		sb2SourcePath = modulesPath.resolve(sb2SourcePath);
+
+		boolean sb1SourceExists = Files.exists(sb1SourcePath);
+		boolean sb2SourceExists = Files.exists(sb2SourcePath);
+		
+		Assert.assertFalse(sb1SourceExists && sb2SourceExists);
+		
+		args = new String[] {"--base", mavenworkspace.getPath(), "buildService"};
+
+		BladeTestResults results = TestUtil.runBlade(mavenworkspace, _extensionsDir, args);
+
+		sb1SourceExists = Files.exists(sb1SourcePath);
+		sb2SourceExists = Files.exists(sb2SourcePath);
+		
+		Assert.assertTrue(sb1SourceExists && sb2SourceExists);
+		
+		String output = results.getOutput();
+		
+		String errors = results.getErrors();
+
+		
+		StringBuilder errorMessage = new StringBuilder();
+
+		if (!output.contains("BUILD SUCCESS")) {
+			errorMessage.append("Expected 'BUILD SUCCESS' message." + System.lineSeparator());
+		}
+
+		if (!errors.isEmpty()) {
+			errorMessage.append(errors + System.lineSeparator());
+		}
+		
+		Assert.assertTrue(errorMessage.toString(), errorMessage.length() == 0);
```

Comment:
Add an assert that will assert the path of a generated file (like something in -api folders)